plugins/cost-insights/src/utils/loading.ts

```diff
@@ -35,7 +35,6 @@ export enum DefaultLoadingAction {
 export const INITIAL_LOADING_ACTIONS = [
   DefaultLoadingAction.UserGroups,
   DefaultLoadingAction.CostInsightsInitial,
-  DefaultLoadingAction.CostInsightsProducts,
```

Comment:
Strictly removing this is incorrect, perhaps LoadingProvider needs to check the config and remove it when no products exist?

```ts
const { products } = useConfig();
const actions = INITIAL_LOADING_ACTIONS.filter(action => products.length || action !== DefaultLoadingAction.CostInsightsProducts);
```

Reply:
Thank you for the comment. Is it not enough that we call `dispatchLoadingProducts(true)`

```ts
try {
  dispatchLoadingProducts(true);
  const responses = await Promise.allSettled(
```

inside ProductInsights component? So we don't need to add the products loading action to `INITIAL_LOADING_ACTIONS` at all, because it is controlled by the ProductInsights component.

Reply:
The purpose of `INITIAL_LOADING_ACTIONS` is to avoid any backdrop flickering while components are rendering. It's true that ProductInsights will control the product loading state after it's rendered, but I think you'll risk corner-case flashes without the LoadingProvider having an initial entry for it.

Reply:
Right, I see your point. I've added a filtering to `LoadingProvider` as you suggested. Thanks.